core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
+      await this.clickOn(mobileSaveDraftButton);
+    } else {
+      await this.clickOn(desktopSaveDraftButton);
+    }
+
+    await this.clickOn(commitMessage);
+    await this.type(commitMessage, 'Testing Testing');
+    await this.clickOn(saveDraftConfirmButton);
+    await this.page.waitForFunction('document.readyState === "complete"');
+    showMessage('Exploration is saved successfully.');
+  }
+
+  async publishExploration(): Promise<string | null> {
+    await this.saveDraftExploration();
     if (this.isViewportAtMobileWidth()) {
       await this.page.waitForSelector('.e2e-test-toast-message', {
         visible: true,
       });
       await this.page.waitForSelector('.e2e-test-toast-message', {
         hidden: true,
       });
-      await this.clickOn(mobileChangesDropdown);
+      await this.clickOn(publishButtonDropDown);
       await this.clickOn(mobilePublishButton);
     } else {
-      await this.page.waitForSelector(
-        `${publishExplorationButton}:not([disabled])`
-      );
-      await this.clickOn(publishExplorationButton);
-    }
-    await this.clickOn(explorationTitleInput);
-    await this.type(explorationTitleInput, `${explorationTitle}`);
-    await this.clickOn(explorationGoalInput);
-    await this.type(explorationGoalInput, `${explorationTitle}`);
-    await this.clickOn(explorationCategoryDropdown);
-    await this.clickOn('Algebra');
-    await this.clickOn(saveExplorationChangesButton);
-    await this.clickOn(explorationConfirmPublishButton);
-    await this.page.waitForSelector(explorationIdElement);
-    const explorationIdUrl = await this.page.$eval(
-      explorationIdElement,
-      element => (element as HTMLElement).innerText
-    );
-    const explorationId = explorationIdUrl.replace(/^.*\/explore\//, '');
+      await this.clickOn(desktopPublishButton);
+    }
+    await this.clickOn(confirmPublishButton);
+    await this.page.waitForSelector(closePublishedPopUpButton, {visible: true});
+
+    explorationUrlAfterPublished = await this.page.url();
+    const explorationId = explorationUrlAfterPublished
+      .replace(/^.*\/create\//, '')
+      .replace(/#\/.*/, '');
+
+    await this.clickOn(closePublishedPopUpButton);
+    await this.expectExplorationToBePublish();
 
-    await this.clickOn(closeShareModalButton);
     return explorationId;
   }
 
+  async expectExplorationToBeAccessibleWithTheUrl(
+    expression: string
+  ): Promise<void> {
+    if (expression === 'Yes') {
+      try {
+        await this.page.goto(explorationUrlAfterPublished);
+        showMessage('Exploration is accessible with the URL i.e Published');
+      } catch (error) {
+        throw new Error('The exploration is not public.');
+      }
+    } else {
+      try {
+        await this.page.goto(explorationUrlAfterPublished);
+        throw new Error('The exploration is still public.');
+      } catch (error) {
+        showMessage(
+          'The exploration is not accessible with the URL i.e Deleted'
```

Comment:
Inaccessible doesn't mean deleted, right? E.g. if it's not yet published then the learner will see a 404 too, right?

If that's the case then maybe you'd need to actually check the creator dashboard instead?

Reply:
We don't need to check the creator dashboard as we are verifying it -> 
      await explorationVisitor.expectExplorationToBeAccessibleByUrl();
      await explorationVisitor.deleteExploration();
      await explorationVisitor.expectExplorationToBeNotAccessibleByUrl();

which means if the exploration is accessible earlier and after delete exploration its not, which means its deleted.

Although the message will be corrected.